tests/test_sql_parse.py

```diff
@@ -1,49 +1,115 @@
 import pytest
 from pydbtools.create_temp_table import check_sql
+from pydbtools.utils import replace_temp_database_name_reference
 
-sql_pass1 = """
-with x as (
-SELECT __temp__.c1, db.c2
-FROM __temp__
-INNER JOIN db
-ON __temp__.c1 = db.c2)
-select * from x
+sql1 = """
+with x as (SELECT __TEMP__.y.c1, db.tb.c2
+FROM __TEMP__.y
+INNER JOIN db.tb
+ON __TEMP__.y.c1 = db.tb.c2)
+select * from x;
 """
 
-sql_pass2 = """
-SELECT __temp__.c1, db.c2
-FROM __temp__
+sql2 = """
+SELECT y.c1, db.c2
+FROM __temp__.y as y
 INNER JOIN db
-ON __temp__.c1 = db.c2
+ON y.c1 = db.c2
+where y.c1 == 'Billy'
 """
 
-sql_fail = """
+sql3 = """
 create table x as (
-SELECT __temp__.c1, db.c2
-FROM __temp__
-INNER JOIN db
-ON __temp__.c1 = db.c2)
+SELECT __temp__.y.c1, tb.c2
+FROM __temp__.y
+INNER JOIN tb
+ON __temp__.y.c1 = tb.c2)
 select * from x
 """
 
+sql4 = """
+SELECT y.c1, db.c2
+FROM __temp__.y as y
+INNER JOIN db
+ON y.c1 = db.c2
+where y.c1 == 'Billy';
+SELECT y.c1, db.c2
+FROM __temp__.y as y
+INNER JOIN db
+ON y.c1 = db.c2
+where y.c1 == 'Billy'
+"""
+
+sql5 = """
+SELECT y.c1, db.c2
+FROM "__temp__".y as y
+"""
```

Comment:
@gkelly900 this is currently failing. It _should_ technically be possible to do this but if it is too annoying to implement then maybe we can just raise it as an issue and document that you shouldn't do this.

Reply:
Should just have something to catch this instance and raise an error. No time for anything else.